components/webapps/blacklist-whitelist-service/src/main/java/com/wso2telco/services/bw/Queries.java

```diff
@@ -56,360 +51,262 @@
-				blackListDTO.setApiID(String.valueOf(apiId));
-				blackListDTO.setApiName(apiInfoArray[0] + ":" + apiInfoArray[1] + ":" + apiInfoArray[2]);
-				blackListDTO.setUserID(userID);
-				blackListDTO.setUserMSISDN(msisdnList);
-				blackListDTO.setValidationRegex(blackListReq.getValidationRegex());
-				blackListDTO.setValidationPrefixGroup(blackListReq.getValidationPrefixGroup());
-				blackListDTO.setValidationDigitsGroup(blackListReq.getValidationDigitsGroup());
-
-
-				blackListWhiteListService.blacklist(blackListDTO);
-
-				return Response.status(Response.Status.OK).entity(successMSG.toString()).build();
+		try {
+			Response.Status status = Response.Status.CREATED;
+			APIBlacklistWhitelistResponseDTO responseDTO = blackListWhiteListService.blacklist(blacklistWhitelistDTO);
 
-			} catch (BusinessException ex) {
-				StringBuilder errorMessage = new StringBuilder("{ \"Failed\": { \"messageId\": \"Blacklist Numbers\", \"text\": \"Blacklist Numbers could not be added to the system\", \"variables\": \"" + ex.getErrorType().toString() + "\" } }");
-				return Response.status(Response.Status.BAD_REQUEST).entity(errorMessage.toString()).build();
+			if(responseDTO.getFailed() > 0){
+				status = Response.Status.BAD_REQUEST;
 			}
 
-		} else {
-			return Response.status(Response.Status.BAD_REQUEST).entity(gson.toJson(errorMSG.toString())).build();
-		}
+			return Response.status(status).entity(gson.toJson(responseDTO)).build();
 
+		} catch (BusinessException ex) {
+			StringBuilder errorMessage = new StringBuilder("{ \"error\": \"" + ex.getMessage() + "\"}");
+			return Response.status(Response.Status.BAD_REQUEST).entity(errorMessage.toString()).build();
+		}
 	}
 
+
+
 	private String[] removeNullMsisdnValues(String[] msisdnList) {
 		List<String> list = new ArrayList<String>(Arrays.asList(msisdnList));
 		list.removeAll(Collections.singleton(null));
 		return list.toArray(new String[list.size()]);
 	}
 
 	/**
-	 * Returns the blacklist per API
+	 * Returns the count per app per api
 	 *
 	 * {empty request body}
 	 */
 	@GET
-	@Path("/GetBlacklistPerApi/{apiId}")
-	@Consumes("application/json")
-	@Produces("text/plain")
-	public Response getBlacklistPerApi(@PathParam("apiId") String apiId) {
-		LOG.debug("getBlacklistPerApi Triggerd  apiName :" + apiId);
+	@Path("/count/{app}/{api}")
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces(MediaType.TEXT_PLAIN)
+	public Response getBlacklistPerApi(@PathParam("app") String app, @PathParam("api") String api,
+									   @QueryParam("action") String action, @QueryParam("sp") String sp) {
 
 		Gson gson = new Gson();
 		StringBuilder errorMSG = new StringBuilder();
 		errorMSG.append("{" + "\"Failed\":").append("{").append("\"messageId\":\"").append("Blacklist result")
 				.append("\",").append("\"text\":\"");
 		errorMSG.append("Blacklist numbers could not be retrieved").append("\"" + "}}");
 
-		if (apiId != null) {
+		app = replaceAllWithWildcard(app);
```

Comment:
Avoid parameter modification. instead, create local variable in the scope and assign the modified value